modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2767,24 +2777,27 @@ def broadcast_apply_full_axis(
                     kw["row_lengths"] = get_length_list(
                         axis_len=len(new_index), num_splits=new_partitions.shape[0]
                     )
-                elif (
-                    axis == 1
-                    and self._row_lengths_cache is not None
-                    and len(new_index) == sum(self._row_lengths_cache)
-                ):
-                    kw["row_lengths"] = self._row_lengths_cache
+                elif axis == 1:
+                    if self._row_lengths_cache is not None and len(new_index) == sum(
+                        self._row_lengths_cache
+                    ):
+                        kw["row_lengths"] = self._row_lengths_cache
+                    elif len(new_index) == 1:
+                        kw["row_lengths"] = [1]
```

Comment:
hm, is this always true? I suggest additionally checking for the number of partitions along the axis:
```suggestion
                    elif len(new_index) == 1 and new_partitions.shape[0] == 1:
                        kw["row_lengths"] = [1]
```
otherwise this may hit us hard in case of empty partitions, for example:
```python
def filter_function(df):
    return df.loc[df["a"] == 1]

# mdf:
#     a
# 0   0
# 1   0    <-- part1
# -----
# 2   0
# 3   1    <-- part2
# -----
# 4   0
# 5   0    <-- part3
# ----

res = mdf.apply_full_axis(axis=0, func=filter_func, new_index=[3], keep_partitioning=False)
# res:
#     a
# (empty)   <-- part1
# -----
# 3   1     <-- part2
# -----
# (empty)   <-- part3
# ----
# res.row_lengths has to be: [0, 1, 0] and not [1]
```

Reply:
good catch, fixed